syntax = "proto3";
package cln;

// This file was automatically derived from the JSON-RPC schemas in
// `doc/schemas`. Do not edit this file manually as it would get
// overwritten.

service Plugin {
    rpc Ping (PluginPingRequest) returns (PluginPongReply);
		rpc HodlInvoice(HodlInvoiceRequest) returns (HodlInvoiceResponse) {}
		rpc HodlInvoiceSettle(HodlInvoiceSettleRequest) returns (HodlInvoiceSettleResponse) {}
		rpc HodlInvoiceCancel(HodlInvoiceCancelRequest) returns (HodlInvoiceCancelResponse) {}
		rpc HodlInvoiceLookup(HodlInvoiceLookupRequest) returns (HodlInvoiceLookupResponse) {}
}

message PluginPingRequest {
   string message = 1;
}

message PluginPongReply {
    string message = 1;
}

message HodlInvoiceRequest {
	uint64 amount_msat = 1;
	string description = 2;
	string label = 3;
	optional uint64 expiry = 4;
	optional uint32 cltv = 5;
}

message HodlInvoiceResponse {
	string bolt11 = 1;
	bytes payment_hash = 2;
	bytes payment_secret = 3;
	uint64 expires_at = 4;
	optional string warning_capacity = 5;
	optional string warning_offline = 6;
	optional string warning_deadends = 7;
	optional string warning_private_unused = 8;
	optional string warning_mpp = 9;
}

message HodlInvoiceSettleRequest {
	bytes payment_hash = 1;
}

message HodlInvoiceSettleResponse {
	enum Hodlstate {
		OPEN = 0;
		SETTLED = 1;
		CANCELED = 2;
		ACCEPTED = 3;
	}
	Hodlstate state = 1;
}

message HodlInvoiceCancelRequest {
	bytes payment_hash = 1;
}

message HodlInvoiceCancelResponse {
	enum Hodlstate {
		OPEN = 0;
		SETTLED = 1;
		CANCELED = 2;
		ACCEPTED = 3;
	}
	Hodlstate state = 1;
}

message HodlInvoiceLookupRequest {
	bytes payment_hash = 1;
}

message HodlInvoiceLookupResponse {
	enum Hodlstate {
		OPEN = 0;
		SETTLED = 1;
		CANCELED = 2;
		ACCEPTED = 3;
	}
	Hodlstate state = 1;
	optional uint32 htlc_expiry = 2;
}
